### PR TITLE
fix: package-js task not checked by mergify

### PR DESCRIPTION
**Pull request by Chriscbr**
Adds new methods `addPostBuildJobTask` and `addPostBuildJobCommands` to `BuildWorkflow` that accepts either a task or sequence of commands respectively, and runs them after the "build" job on the main build workflow.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by eladb — CHANGES_REQUESTED** [review]
I think it might be possible to add the integ task as a post build job now to the build workflow.

Something like this:

```ts
project.buildWorkflow.addPostBuildJob('integ', ...);
```

File: .projenrc.js [hunk 1]
```diff
@@ -219,9 +212,19 @@ new github.TaskWorkflow(project.github, {
         'go-version': '^1.16.0',
       },
     },
+    {
+      name: 'Prepare repository',
```
Comment (eladb) [inline .projenrc.js:216]: So hairy... I wonder if we can offer an even higher level abstractions for post build jobs.

Something like:

```ts
buildWorkflow.addPostBuildJobCommand("...", { tools: {} });

```

**Review by eladb — APPROVED** [review]

File: .projenrc.js (hunk 1 above)
Comment (Chriscbr) [reply .projenrc.js:216]: I'll see what I can do

**Comment by Chriscbr** [comment]
Ahhhhhhhhhhh that explains the errors... in each job step the directory you're in gets reset. I forget these things sometimes 🙃

**Comment by Chriscbr** [comment]
> ./projen.bash: line 8: /home/runner/work/projen/projen/dist/bin/projen: Permission denied

It looks like GitHub actions also marks files as un-executable for some reason...

File: src/build/build-workflow.ts [hunk 2]
```diff
@@ -187,6 +188,77 @@ export class BuildWorkflow extends Component {
...
+   * The task will have access to build artifacts if `project.artifactsDirectory`
+   * is defined.
+   *
+   * Jobs are executed _only_ if the build did NOT self mutate. If the build
+   * self-mutate, the branch will either be updated or the build will fail (in
+   * forks), so there is no point in executing the post-build job.
+   *
+   * @param options Specify tools and other options
+   */
+  public addPostBuildJobTask(task: Task, options: AddPostBuildJobTaskOptions) {
```
Comment (eladb) [inline src/build/build-workflow.ts:204]: Can we _also_ add a lower-level `addPostBuildJobCommand()` which will be used by `addPostBuildJobTask()`?

Comment (eladb) [inline src/build/build-workflow.ts:204]: Feels like we should be able to use this in `JsiiProject` to execute the `jsii-pacmak` commands, no?

**Review by eladb — CHANGES_REQUESTED** [review]

File: src/build/build-workflow.ts (hunk 2 above)
Comment (Chriscbr) [reply src/build/build-workflow.ts:205]: Done

Comment (Chriscbr) [reply src/build/build-workflow.ts:205]: This part is a little bit less clear so I'm going to leave this open for later refactoring if that's OK.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -186,3 +186,29 @@ jobs:
...
+      - name: Download build artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: build-artifact
+          path: dist
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```
Comment (Chriscbr) [inline .github/workflows/build.yml:211]: I don't think it matters which operation goes first (downloading build artifacts or checking out the repo?) but I could be wrong

**Comment by eladb** [comment]
## Please read this to avoid merge hell

Hola @Chriscbr, apologies for hassle. I've enabled Prettify as part of #1453. This means that most of the source files have changed and you will likely see quite a lot of merge conflicts when you try to merge from `main`.

*But there's an easy way to handle this.*

First, merge your branch from this tag:

```sh
$ git fetch origin
$ git merge 1020d0ad3f95bb4e1201f8640d440ffb43ccf230
```

Edit the `.projenrc.js` file in your branch and add this to **line 57** (right
after `codeCov: true`):

```js
  prettier: true,
```

Now, run:

```sh
$ git clean -fdx .
$ yarn
$ bash projen.bash
$ yarn eslint
$ git add .
$ git commit -m 'format code'
```

This will reformat all the code in your branch to match `main`.

Now, merge from main:

```sh
$ git merge origin/main
```

File: src/build/build-workflow.ts (hunk 2 above)
Comment (eladb) [reply src/build/build-workflow.ts:204]: Sure.

**Review by eladb — APPROVED** [review]

**Comment by Chriscbr** [comment]
~magic incantations~ merge instructions worked as expected 👍

**Comment by moltar** [comment]
I think this PR _might be_ causing this:

```diff
diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index 49af2d1..4d90af6 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -24,7 +24,7 @@ jobs:
         with:
           node-version: 16.13.0
       - name: Install dependencies
-        run: npm ci
+        run: npm install
       - name: build
         run: npx projen build
       - name: Turbo Build

```

Not sure if this is an intended consequence. But I think we should be always using `npm ci` in GitHub Actions?

**Comment by eladb** [comment]
> I think this PR _might be_ causing this:
> 
> 
> 
> ```diff
> 
> diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
> 
> index 49af2d1..4d90af6 100644
> 
> --- a/.github/workflows/build.yml
> 
> +++ b/.github/workflows/build.yml
> 
> @@ -24,7 +24,7 @@ jobs:
> 
>          with:
> 
>            node-version: 16.13.0
> 
>        - name: Install dependencies
> 
> -        run: npm ci
> 
> +        run: npm install
> 
>        - name: build
> 
>          run: npx projen build
> 
>        - name: Turbo Build
> 
> 
> 
> ```
> 
> 
> 
> Not sure if this is an intended consequence. But I think we should be always using `npm ci` in GitHub Actions?

This is the PR that changed this behavior: https://github.com/projen/projen/pull/1467 and it was intentional.

**Pull request merged**